sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/HttpResponse.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.javasdk;
+
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class HttpResponse {
+  private final int statusCode;
+  private final Optional<String> contentType;
+  private final byte[] body;
+
+  //TODO can't make it private because we use it internally for deserialization
+  public HttpResponse(int statusCode, Optional<String> contentType, byte[] body) {
```

Comment:
I think it's safe to expose this constructor with a basic validation like the one below.